api/v1alpha1/applicationvolumereplication_types.go

```diff
@@ -65,57 +73,53 @@ type ApplicationVolumeReplicationSpec struct {
 	// and AWS_SECRET_ACCESS_KEY.  The value of this field, will be progated to every VRG.
 	// See VRG spec for more details.
 	S3SecretName string `json:"s3SecretName"`
+
+	// Action is either failover or failback operation
+	Action DRAction `json:"action,omitempty"`
 }
 
-// DRState for each subscription
+// DRState for keeping track of the DR placement
 // +kubebuilder:validation:Enum=Initial;Failing-over;Failed-over;Failing-back;Failed-back
 type DRState string
 
 // These are the valid values for DRState
 const (
 	// Initial, this is the state that will be recorded in the AVR status
-	// when initial deplyment has been performed successfully (per subscription)
+	// when initial deplyment has been performed successfully
 	Initial DRState = "Initial"
 
-	// FailingOver, state recorded in the AVR status per subscription when the failover
+	// FailingOver, state recorded in the AVR status when the failover
 	// is initiated but has not been completed yet
 	FailingOver DRState = "Failing-over"
 
-	// FailedOver, state recorded in the AVR status per subscription when the failover
+	// FailedOver, state recorded in the AVR status when the failover
 	// process has completed
 	FailedOver DRState = "Failed-over"
 
-	// FailingBack, state recorded in the AVR status per subscription when the failback
+	// FailingBack, state recorded in the AVR status when the failback
 	// is initiated but has not been completed yet
 	FailingBack DRState = "Failing-back"
 
-	// FailedBack, state recorded in the AVR status per subscription when the failback
+	// FailedBack, state recorded in the AVR status when the failback
 	// process has completed
 	FailedBack DRState = "Failed-back"
-)
-
-// LastKnownDRStateMap defines per subscription the last state, key is subscription name
-type LastKnownDRStateMap map[string]DRState
 
-// SubscriptionPlacementDecision lists each subscription with its home and peer clusters
-type SubscriptionPlacementDecision struct {
-	HomeCluster     string `json:"homeCluster,omitempty"`
-	PeerCluster     string `json:"peerCluster,omitempty"`
-	PrevHomeCluster string `json:"prevHomeCluster,omitempty"`
-}
-
-// SubscriptionPlacementDecisionMap defines per subscription placement decision, key is subscription name
-type SubscriptionPlacementDecisionMap map[string]*SubscriptionPlacementDecision
+	// TODO: implement the relocation
+	Relocating DRState = "Relocating"
+	Relocated  DRState = "Relocated"
+)
 
 // ApplicationVolumeReplicationStatus defines the observed state of ApplicationVolumeReplication
 type ApplicationVolumeReplicationStatus struct {
-	Decisions         SubscriptionPlacementDecisionMap `json:"decisions,omitempty"`
-	LastKnownDRStates LastKnownDRStateMap              `json:"lastKnownDRStates,omitempty"`
-	LastUpdateTime    metav1.Time                      `json:"lastUpdateTime"`
+	PreferredDecision plrv1.PlacementDecision `json:"decision,omitempty"`
+	ClusterToClean    string                  `json:"clusterToClean,omitempty"`
```

Comment:
We should dynamically determine this value, based on current spec (say failover), then we may have pending actions on cleanup on all non-failoverClusters from DRPeers list. (linked comment in `ensureCleanup` function)

Reply:
Done